nsdb-cluster/src/main/scala/io/radicalbit/nsdb/cluster/actor/AbstractClusterListener.scala

```diff
@@ -109,12 +109,12 @@ abstract class AbstractClusterListener extends Actor with ActorLogging with Futu
   protected def createNodeActorsGuardian(): ActorRef = {
     context.system.actorOf(
       NodeActorsGuardian.props(self, nodeId).withDeploy(Deploy(scope = RemoteScope(cluster.selfMember.address))),
-      name = s"guardian_${selfNodeName}"
+      name = s"guardian_${nodeId}_${selfNodeName}"
```

Comment:
out of curiosity. may two nodes share the same name?

Reply:
It's highly unlikely. The node id is a Java UUID stored inside the metadata folder of the node. The `selfNodeName` is a concatenation of the address and the port.
If a node is restarted in a cloud env for instance, it would be assigned to another address.
The only corner case, maybe, is a backup and restore process. 
In case of a backup, we must ensure that the node id will not be included.